pkg/controllers/sidecar/identity/member.go

```diff
@@ -63,27 +66,43 @@ func Retrieve(ctx context.Context, name, namespace string, kubeclient kubernetes
 	}, nil
 }
 
-func (m *Member) GetSeeds(ctx context.Context, kubeClient kubernetes.Interface) ([]string, error) {
-	var services *corev1.ServiceList
-	var err error
+// GetSeed return a single node contact point used for bootstrapping.
+// It is enough to return a single node because bootstrap process will fail if any of the nodes in cluster is down.
+func (m *Member) GetSeed(ctx context.Context, coreClient v1.CoreV1Interface) (string, error) {
```

Comment:
If all nodes have to be UN for a new one to join, would it work if we just return the fixed DNS name for first pod in the statefulset?
If it has to be a fixed "external" IP would a fixed service for the first node work?

Reply:
That's what I tried at the beginning, but it doesn't work. DNS resolves to PodIP on the same pod, and to service clusterIP on other pods. Scylla cannot bootstrap using PodIP, it has to use one of the node broadcast address. 
Fixed IP also means that fixed node is not going to be replacable since it would try to connect to itself. 
To avoid those problems why I pick up any UN node, or itself (serviceIP) if it is the first node in the cluster.

Reply:
hm, the service DNS resolves to ClusterIP for me
```
[root@basic-us-east-1-us-east-1a-0 /]# ping basic-us-east-1-us-east-1a-0.test.svc
PING basic-us-east-1-us-east-1a-0.test.svc.cluster.local (10.97.130.36) 56(84) bytes of data.
...
[root@basic-us-east-1-us-east-1a-0 /]# ip a
...
12: eth0@if13: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
    link/ether 02:42:ac:11:00:06 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 172.17.0.6/16 brd 172.17.255.255 scope global eth0
```
you are right about replacing itself - i guess we could just get all of the node service names there